lib/unexpectedMitm.js

```diff
@@ -1151,28 +1157,55 @@ module.exports = {
                                     }).then(function () {
                                         return getMockResponse(remainingDescription.response);
                                     }).spread(function (mockResponse, mockResponseError) {
-                                        httpConversationSatisfySpec.exchanges.push({
+                                        var spec = {
                                             request: expectedRequestProperties,
                                             response: mockResponseError || trimMockResponse(mockResponse)
-                                        });
+                                        };
+
+                                        exchangeParts.push({ spec: spec });
 
                                         return nextItem();
-                                    }).caught(reject);
+                                    });
                                 } else {
-                                    resolve([fulfilmentValue, httpConversation, httpConversationSatisfySpec]);
+                                    throw new UnexercisedMocksError();
                                 }
                             })();
                         } else {
-                            resolve([fulfilmentValue, httpConversation, httpConversationSatisfySpec]);
+                            resolve([fulfilmentValue]);
                         }
                     }).caught(function (e) {
-                        if (__lastError) {
-                            reject(__lastError);
+                        exchangeParts.push(e);
+                        reject(e);
+                    });
+                }).catch(function () {
+                    var lastPartOrError = exchangeParts[exchangeParts.length - 1];
```

Comment:
Will `exchangeParts.length` always be > 0 here?

Reply:
Good question - I'll instrument the code to check. This might be relying on the return undefined behaviour but if so I think an explicit check might be better.

Reply:
I realize that it'll still come out right even when dereferencing `-1` because of the truthiness check below, but I'd rather not go out of bounds. Call me a freak :)

Reply:
The diff changed but it's addressed this yet - this is taken care of in the work that pushes this in the direction of separated mocker.